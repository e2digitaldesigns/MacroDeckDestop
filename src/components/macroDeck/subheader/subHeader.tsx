import React from "react";
import * as Styled from "./subHeader.style";
import { ChevronRight } from "react-bootstrap-icons";
import BreadCrumb from "./breadCrumb/breadCrumb";
import {
  useActions,
  useAppData,
  useButton,
  usePage,
  useProfile
} from "../../../hooks";
import { BreadCrumbMenuTypes, allActionMap } from "../../../types";

interface IntBreadWrapper {
  activeBreadCrumbMenu: BreadCrumbMenuTypes | null;
  dropDownType: BreadCrumbMenuTypes;
  label: string;
  showChev?: boolean;
  handleSetActiveBreadCrumbMenu: (
    activeMenu: BreadCrumbMenuTypes | null
  ) => void;
  title: string;
}

const BreadWrapper: React.FC<IntBreadWrapper> = ({
  activeBreadCrumbMenu,
  dropDownType,
  label,
  showChev = true,
  handleSetActiveBreadCrumbMenu,
  title
}) => {
  return (
    <>
      {showChev && (
        <Styled.BreadCrumbHolder>
          <ChevronRight />
        </Styled.BreadCrumbHolder>
      )}
      <BreadCrumb
        activeBreadCrumbMenu={activeBreadCrumbMenu}
        dropDownType={dropDownType}
        label={label}
        title={title}
        handleSetActiveBreadCrumbMenu={handleSetActiveBreadCrumbMenu}
      />
    </>
  );
};

export const MacroDeckSubHeader: React.FC = () => {
  const [activeBreadCrumbMenu, setActiveBreadCrumbMenu] =
    React.useState<BreadCrumbMenuTypes | null>(BreadCrumbMenuTypes.Page);
  const { readPage, pageCount } = usePage();
  const { readProfile } = useProfile();
  const { appState } = useAppData();
  const { getActiveButton } = useButton();
  const { getActiveAction } = useActions();

  const profile = readProfile();
  const page = readPage();
  const pagesCount = pageCount();
  const activeButton = getActiveButton();
  const activeAction = getActiveAction();

  const profileTitle = profile?.profileName || "Select a Profile";
  const pageNumber = page?.number || "0";
  const pageTitle = `${pageNumber} / ${pagesCount}`;

  const buttonPadCount = profile?.buttonPads || 0;
  const buttonPadNum = activeButton?.buttonPadNum || 0;
  const buttonPadTitle = `${buttonPadNum} / ${buttonPadCount}`;

  let actionTitle = activeAction?.subAction || activeAction?.action || null;
  actionTitle = actionTitle ? allActionMap[actionTitle] : "No Action";

  React.useEffect(() => {
    setActiveBreadCrumbMenu(activeBreadCrumbMenu => null);
  }, [appState.active]);

  const handleSetActiveBreadCrumbMenu = (
    activeMenu: BreadCrumbMenuTypes | null
  ) => {
    setActiveBreadCrumbMenu(activeMenu);
  };

  return (
    <Styled.SubHeader>
      <BreadWrapper
        activeBreadCrumbMenu={activeBreadCrumbMenu}
        dropDownType={BreadCrumbMenuTypes.Profile}
        showChev={false}
        label="profile"
        handleSetActiveBreadCrumbMenu={handleSetActiveBreadCrumbMenu}
        title={profileTitle}
      />

      <BreadWrapper
        activeBreadCrumbMenu={activeBreadCrumbMenu}
        dropDownType={BreadCrumbMenuTypes.Page}
        label="page"
        handleSetActiveBreadCrumbMenu={handleSetActiveBreadCrumbMenu}
        title={pageTitle}
      />

      <BreadWrapper
        activeBreadCrumbMenu={activeBreadCrumbMenu}
        dropDownType={BreadCrumbMenuTypes.ButtonPad}
        label="button pad"
        handleSetActiveBreadCrumbMenu={handleSetActiveBreadCrumbMenu}
        title={buttonPadTitle}
      />

      <BreadWrapper
        activeBreadCrumbMenu={activeBreadCrumbMenu}
        dropDownType={BreadCrumbMenuTypes.Action}
        label="action"
        handleSetActiveBreadCrumbMenu={handleSetActiveBreadCrumbMenu}
        title={actionTitle}
      />
    </Styled.SubHeader>
  );
};
