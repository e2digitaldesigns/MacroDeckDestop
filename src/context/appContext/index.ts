/* istanbul ignore file */
export * from "./appContext";
export * from "./defaultState";
