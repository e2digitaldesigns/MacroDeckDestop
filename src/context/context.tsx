import React from "react";
import { TemplateContextProvider } from "./templateContext";
import { ThemeContextProvider } from "./themeContext";

export interface IThemeComponents {
  TemplateProvider: React.ElementType;
  ThemeProvider: React.ElementType;
}

export const ApplicationContext: React.FC & IThemeComponents = () => <div />;

ApplicationContext.TemplateProvider = TemplateContextProvider;
ApplicationContext.ThemeProvider = ThemeContextProvider;
