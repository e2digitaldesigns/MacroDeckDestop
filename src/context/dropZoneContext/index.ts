export * from "./dropZoneContext";
export * from "./defaultState";
