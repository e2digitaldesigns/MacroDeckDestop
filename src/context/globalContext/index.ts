export * from "./globalContext";
export * from "./defaultState";
