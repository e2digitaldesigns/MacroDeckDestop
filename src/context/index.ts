export * from "./appContext";
export * from "./dropZoneContext";
export * from "./globalContext";
export * from "./templateContext";
export * from "./themeContext";
