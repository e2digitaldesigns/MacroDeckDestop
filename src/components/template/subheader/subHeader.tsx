import React from "react";
import * as Styled from "./subHeader.style";

const SubHeader: React.FC = () => {
  return <Styled.SubHeader></Styled.SubHeader>;
};

export default SubHeader;
