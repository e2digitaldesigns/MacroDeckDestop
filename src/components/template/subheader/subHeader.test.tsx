import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { TemplateSubHeader } from "./subHeader";

import { ApplicationContext } from "../../../context/context";

const testSetup = () => {
  return render(
    <ApplicationContext.ThemeProvider>
      <TemplateSubHeader />
    </ApplicationContext.ThemeProvider>
  );
};

describe("<Template SubHeader/>", () => {
  const wrapper = testSetup();
  it("Should render without errors", () => {
    const component = wrapper;
    expect(component).toMatchSnapshot();
  });
});
