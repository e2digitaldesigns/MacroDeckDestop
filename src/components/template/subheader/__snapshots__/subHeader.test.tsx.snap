// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Template SubHeader/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bczRLJ fbMjRA"
      >
        <div
          class="sc-eCYdqJ kbsUPd"
        >
          <div
            class="sc-jSMfEi drGKcz"
          >
            <div
              class="sc-gKXOVf kBJLFY"
            >
              profile
            </div>
            <div
              class="sc-iBkjds fSRmgS"
            >
              <div
                class="sc-ftvSup fDuCrV"
                width="70"
              >
                Select a Profile
              </div>
              <div
                class="sc-papXJ fKTrfV"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-gsnTZi bvKKze"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="sc-eCYdqJ kbsUPd"
        >
          <div
            class="sc-jSMfEi drGKcz"
          >
            <div
              class="sc-gKXOVf kBJLFY"
            >
              page
            </div>
            <div
              class="sc-iBkjds fSRmgS"
            >
              <div
                class="sc-ftvSup kwujYm"
                width="55"
              >
                0 / 0
              </div>
              <div
                class="sc-papXJ fKTrfV"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-gsnTZi bvKKze"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="sc-eCYdqJ kbsUPd"
        >
          <div
            class="sc-jSMfEi drGKcz"
          >
            <div
              class="sc-gKXOVf kBJLFY"
            >
              button pad
            </div>
            <div
              class="sc-iBkjds fSRmgS"
            >
              <div
                class="sc-ftvSup kwujYm"
                width="55"
              >
                0 / 0
              </div>
              <div
                class="sc-papXJ fKTrfV"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-gsnTZi bvKKze"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="sc-eCYdqJ kbsUPd"
        >
          <div
            class="sc-jSMfEi drGKcz"
          >
            <div
              class="sc-gKXOVf kBJLFY"
            >
              action
            </div>
            <div
              class="sc-iBkjds fSRmgS"
            >
              <div
                class="sc-ftvSup ljeEOI"
                width="60"
              >
                no action selected
              </div>
              <div
                class="sc-papXJ fKTrfV"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-bczRLJ fbMjRA"
    >
      <div
        class="sc-eCYdqJ kbsUPd"
      >
        <div
          class="sc-jSMfEi drGKcz"
        >
          <div
            class="sc-gKXOVf kBJLFY"
          >
            profile
          </div>
          <div
            class="sc-iBkjds fSRmgS"
          >
            <div
              class="sc-ftvSup fDuCrV"
              width="70"
            >
              Select a Profile
            </div>
            <div
              class="sc-papXJ fKTrfV"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gsnTZi bvKKze"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="sc-eCYdqJ kbsUPd"
      >
        <div
          class="sc-jSMfEi drGKcz"
        >
          <div
            class="sc-gKXOVf kBJLFY"
          >
            page
          </div>
          <div
            class="sc-iBkjds fSRmgS"
          >
            <div
              class="sc-ftvSup kwujYm"
              width="55"
            >
              0 / 0
            </div>
            <div
              class="sc-papXJ fKTrfV"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gsnTZi bvKKze"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="sc-eCYdqJ kbsUPd"
      >
        <div
          class="sc-jSMfEi drGKcz"
        >
          <div
            class="sc-gKXOVf kBJLFY"
          >
            button pad
          </div>
          <div
            class="sc-iBkjds fSRmgS"
          >
            <div
              class="sc-ftvSup kwujYm"
              width="55"
            >
              0 / 0
            </div>
            <div
              class="sc-papXJ fKTrfV"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gsnTZi bvKKze"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="sc-eCYdqJ kbsUPd"
      >
        <div
          class="sc-jSMfEi drGKcz"
        >
          <div
            class="sc-gKXOVf kBJLFY"
          >
            action
          </div>
          <div
            class="sc-iBkjds fSRmgS"
          >
            <div
              class="sc-ftvSup ljeEOI"
              width="60"
            >
              no action selected
            </div>
            <div
              class="sc-papXJ fKTrfV"
            >
              <svg
                fill="currentColor"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.247 11.14 2.451 5.658C1.885 5.013 2.345 4 3.204 4h9.592a1 1 0 0 1 .753 1.659l-4.796 5.48a1 1 0 0 1-1.506 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
